hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueService.kt

```diff
@@ -57,27 +57,35 @@ open class HmppsQueueService(
       .map { retryDlqRequest -> retryDlqMessages(retryDlqRequest) }
 
   private suspend fun HmppsQueue.retryDlqMessages(): RetryDlqResult {
-    if (sqsDlqClient == null || dlqUrl == null) return RetryDlqResult(0, listOf())
+    if (sqsDlqClient == null || dlqUrl == null) {
+      return RetryDlqResult(0)
+    }
 
     val messageCount = sqsDlqClient.countMessagesOnQueue(dlqUrl!!).await()
-    val map: Map<String, Any> = HashMap()
-
-    val messages = (1..messageCount)
-      .asFlow()
-      .map { sqsDlqClient.receiveMessage(ReceiveMessageRequest.builder().queueUrl(dlqUrl).maxNumberOfMessages(1).messageAttributeNames("All").build()) }
-      .mapNotNull { it.await().messages().firstOrNull() }
-      .map { msg ->
-        sqsClient.sendMessage(SendMessageRequest.builder().queueUrl(queueUrl).messageBody(msg.body()).messageAttributes(msg.messageAttributes()).build()).await()
-        sqsDlqClient.deleteMessage(DeleteMessageRequest.builder().queueUrl(dlqUrl).receiptHandle(msg.receiptHandle()).build()).await()
-        DlqMessage(messageId = msg.messageId(), body = gson.fromJson(msg.body(), map.javaClass))
-      }.toList()
 
     if (messageCount > 0) {
-      log.info("For dlq ${this.dlqName} we found $messageCount messages, attempted to retry ${messages.size}")
-      telemetryClient?.trackEvent("RetryDLQ", mapOf("dlq-name" to dlqName, "messages-found" to "$messageCount", "messages-retried" to "${messages.size}"), null)
+      val dlqArn = sqsDlqClient.getQueueAttributes(
+        GetQueueAttributesRequest.builder().queueUrl(dlqUrl).attributeNames(QueueAttributeName.QUEUE_ARN).build(),
+      ).await().attributes()[QueueAttributeName.QUEUE_ARN]
+
+      val arn = sqsClient.getQueueAttributes(
+        GetQueueAttributesRequest.builder().queueUrl(queueUrl).attributeNames(QueueAttributeName.QUEUE_ARN).build(),
+      ).await().attributes()[QueueAttributeName.QUEUE_ARN]
```

Comment:
I wonder if we should add this to `HmppsQueue` [here](https://github.com/ministryofjustice/hmpps-spring-boot-sqs/blob/main/hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueue.kt#L13)? Seems better than 2 extra requests every retryDLQ.

Reply:
Good point